File: MiniDOMOffice/Office/Veicolo/VeicoliCursor.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Class Office


    ''' <summary>
    ''' Cursore sulla tabella dei veicoli
    ''' </summary>
    ''' <remarks></remarks>
    Public Class VeicoliCursor
        Inherits DBObjectCursorPO(Of Veicolo)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Tipo As New CCursorFieldObj(Of String)("Tipo")
        Private m_Modello As New CCursorFieldObj(Of String)("Modello")
        Private m_Seriale As New CCursorFieldObj(Of String)("Seriale")
        Private m_Alimentazione As New CCursorFieldObj(Of String)("Alimentazione")
        Private m_KmALitro As New CCursorField(Of Single)("KmALitro")
        Private m_IconURL As New CCursorFieldObj(Of String)("IconURL")
        Private m_DataAcquisto As New CCursorField(Of Date)("DataAcquisto")
        Private m_DataDismissione As New CCursorField(Of Date)("DataDismissione")
        Private m_StatoVeicolo As New CCursorField(Of StatoVeicolo)("StatoVeicolo")
        Private m_Targa As New CCursorFieldObj(Of String)("Targa")
        Private m_DataImmatricolazione As New CCursorField(Of Date)("DataImmatricolazione")
        Private m_ConsumoUrbano As New CCursorField(Of Single)("ConsumoUrbano")
        Private m_ConsumoExtraUrbano As New CCursorField(Of Single)("ConsumoExtraUrbano")
        Private m_ConsumoCombinato As New CCursorField(Of Single)("ConsumoCombinato")
        Private m_Note As New CCursorFieldObj(Of String)("Note")
        Private m_Flags As New CCursorField(Of VeicoloFlags)("Flags")
        Private m_IDProprietario As New CCursorField(Of Integer)("IDProprietario")
        Private m_NomeProprietario As New CCursorFieldObj(Of String)("NomeProprietario")

        Public Sub New()
        End Sub

        Public ReadOnly Property IDProprietario As CCursorField(Of Integer)
            Get
                Return Me.m_IDProprietario
            End Get
        End Property

        Public ReadOnly Property NomeProprietario As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeProprietario
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of VeicoloFlags)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property KmALitro As CCursorField(Of Single)
            Get
                Return Me.m_KmALitro
            End Get
        End Property

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property Tipo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Tipo
            End Get
        End Property

        Public ReadOnly Property Modello As CCursorFieldObj(Of String)
            Get
                Return Me.m_Modello
            End Get
        End Property

        Public ReadOnly Property Seriale As CCursorFieldObj(Of String)
            Get
                Return Me.m_Seriale
            End Get
        End Property

        Public ReadOnly Property Alimentazione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Alimentazione
            End Get
        End Property

        Public ReadOnly Property IconURL As CCursorFieldObj(Of String)
            Get
                Return Me.m_IconURL
            End Get
        End Property

        Public ReadOnly Property DataAcquisto As CCursorField(Of Date)
            Get
                Return Me.m_DataAcquisto
            End Get
        End Property

        Public ReadOnly Property DataDismissione As CCursorField(Of Date)
            Get
                Return Me.m_DataDismissione
            End Get
        End Property

        Public ReadOnly Property StatoVeicolo As CCursorField(Of StatoVeicolo)
            Get
                Return Me.m_StatoVeicolo
            End Get
        End Property

        Public ReadOnly Property Targa As CCursorFieldObj(Of String)
            Get
                Return Me.m_Targa
            End Get
        End Property

        Public ReadOnly Property DataImmatricolazione As CCursorField(Of Date)
            Get
                Return Me.m_DataImmatricolazione
            End Get
        End Property

        Public ReadOnly Property ConsumoUrbano As CCursorField(Of Single)
            Get
                Return Me.m_ConsumoUrbano
            End Get
        End Property

        Public ReadOnly Property ConsumoExtraUrbano As CCursorField(Of Single)
            Get
                Return Me.m_ConsumoExtraUrbano
            End Get
        End Property

        Public ReadOnly Property ConsumoCombinato As CCursorField(Of Single)
            Get
                Return Me.m_ConsumoCombinato
            End Get
        End Property

        Public ReadOnly Property Note As CCursorFieldObj(Of String)
            Get
                Return Me.m_Note
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return Office.Veicoli.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeVeicoli"
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function
    End Class



End Class

File: MiniDOMOffice/Office/Veicolo/VeicoliCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {


        /// <summary>
        /// Cursore sulla tabella dei veicoli
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class VeicoliCursor 
            : minidom.Databases.DBObjectCursorPO<Veicolo>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_Tipo = new DBCursorStringField("Tipo");
            private DBCursorStringField m_Modello = new DBCursorStringField("Modello");
            private DBCursorStringField m_Seriale = new DBCursorStringField("Seriale");
            private DBCursorStringField m_Alimentazione = new DBCursorStringField("Alimentazione");
            private DBCursorField<double> m_KmALitro = new DBCursorField<double>("KmALitro");
            private DBCursorStringField m_IconURL = new DBCursorStringField("IconURL");
            private DBCursorField<DateTime> m_DataAcquisto = new DBCursorField<DateTime>("DataAcquisto");
            private DBCursorField<DateTime> m_DataDismissione = new DBCursorField<DateTime>("DataDismissione");
            private DBCursorField<StatoVeicolo> m_StatoVeicolo = new DBCursorField<StatoVeicolo>("StatoVeicolo");
            private DBCursorStringField m_Targa = new DBCursorStringField("Targa");
            private DBCursorField<DateTime> m_DataImmatricolazione = new DBCursorField<DateTime>("DataImmatricolazione");
            private DBCursorField<double> m_ConsumoUrbano = new DBCursorField<double>("ConsumoUrbano");
            private DBCursorField<double> m_ConsumoExtraUrbano = new DBCursorField<double>("ConsumoExtraUrbano");
            private DBCursorField<double> m_ConsumoCombinato = new DBCursorField<double>("ConsumoCombinato");
            private DBCursorStringField m_Note = new DBCursorStringField("Note");
            private DBCursorField<VeicoloFlags> m_Flags = new DBCursorField<VeicoloFlags>("Flags");
            private DBCursorField<int> m_IDProprietario = new DBCursorField<int>("IDProprietario");
            private DBCursorStringField m_NomeProprietario = new DBCursorStringField("NomeProprietario");

            /// <summary>
            /// Costruttore
            /// </summary>
            public VeicoliCursor()
            {
            }

            /// <summary>
            /// IDProprietario
            /// </summary>
            public DBCursorField<int> IDProprietario
            {
                get
                {
                    return m_IDProprietario;
                }
            }

            /// <summary>
            /// NomeProprietario
            /// </summary>
            public DBCursorStringField NomeProprietario
            {
                get
                {
                    return m_NomeProprietario;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<VeicoloFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// KmALitro
            /// </summary>
            public DBCursorField<double> KmALitro
            {
                get
                {
                    return m_KmALitro;
                }
            }

            /// <summary>
            /// Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// Tipo
            /// </summary>
            public DBCursorStringField Tipo
            {
                get
                {
                    return m_Tipo;
                }
            }

            /// <summary>
            /// Modello
            /// </summary>
            public DBCursorStringField Modello
            {
                get
                {
                    return m_Modello;
                }
            }

            /// <summary>
            /// Seriale
            /// </summary>
            public DBCursorStringField Seriale
            {
                get
                {
                    return m_Seriale;
                }
            }

            /// <summary>
            /// Alimentazione
            /// </summary>
            public DBCursorStringField Alimentazione
            {
                get
                {
                    return m_Alimentazione;
                }
            }

            /// <summary>
            /// IconURL
            /// </summary>
            public DBCursorStringField IconURL
            {
                get
                {
                    return m_IconURL;
                }
            }

            /// <summary>
            /// DataAcquisto
            /// </summary>
            public DBCursorField<DateTime> DataAcquisto
            {
                get
                {
                    return m_DataAcquisto;
                }
            }

            /// <summary>
            /// DataDismissione
            /// </summary>
            public DBCursorField<DateTime> DataDismissione
            {
                get
                {
                    return m_DataDismissione;
                }
            }

            /// <summary>
            /// StatoVeicolo
            /// </summary>
            public DBCursorField<StatoVeicolo> StatoVeicolo
            {
                get
                {
                    return m_StatoVeicolo;
                }
            }

            /// <summary>
            /// Targa
            /// </summary>
            public DBCursorStringField Targa
            {
                get
                {
                    return m_Targa;
                }
            }

            /// <summary>
            /// DataImmatricolazione
            /// </summary>
            public DBCursorField<DateTime> DataImmatricolazione
            {
                get
                {
                    return m_DataImmatricolazione;
                }
            }

            /// <summary>
            /// ConsumoUrbano
            /// </summary>
            public DBCursorField<double> ConsumoUrbano
            {
                get
                {
                    return m_ConsumoUrbano;
                }
            }

            /// <summary>
            /// ConsumoExtraUrbano
            /// </summary>
            public DBCursorField<double> ConsumoExtraUrbano
            {
                get
                {
                    return m_ConsumoExtraUrbano;
                }
            }

            /// <summary>
            /// ConsumoCombinato
            /// </summary>
            public DBCursorField<double> ConsumoCombinato
            {
                get
                {
                    return m_ConsumoCombinato;
                }
            }

            /// <summary>
            /// Note
            /// </summary>
            public DBCursorStringField Note
            {
                get
                {
                    return m_Note;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.Veicoli;
            }

          
        }
    }
}